app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/ProfileController.scala

```diff
@@ -68,12 +70,44 @@ class ProfileController @Inject() (
       }
   }
 
-  def doesProfileExist(eori: String): Action[AnyContent] = identify.async {
+  def doesProfileExist(eori: String): Action[AnyContent] = identify.async { implicit request =>
     profileRepository
       .get(eori)
-      .map {
-        case Some(_) => Ok
-        case None    => NotFound
+      .flatMap {
+        case Some(_) => Future.successful(Ok)
+        case None    =>
+          customsDataStoreConnector.getEoriHistory(eori).flatMap {
+            case Some(eoriHistoryResponse) =>
+              if (eoriHistoryResponse.eoriHistory.isEmpty) Future.successful(NotFound)
+              else {
+
+                val latestEoriResult = profileRepository.get(eoriHistoryResponse.eoriHistory.head.eori).flatMap {
```

Comment:
this doesn't need to be in a val (nice to have)

Reply:
We are checking the results later https://github.com/hmrc/trader-goods-profiles-data-store/pull/70/files#diff-113d27ac3d5fd1b509571f7d992ab8473dc3ef0e2fc1f32bc5d245f8e1cd6bcdR104